src/wp-includes/fonts.php

```diff
@@ -107,31 +125,70 @@ function wp_unregister_font_collection( string $slug ) {
  *     @type string|false $error   False or error message.
  * }
  */
-function wp_get_font_dir() {
+function wp_font_dir( $create_dir = true ) {
+	/*
+	 * Allow extenders to manipulate the font directory consistently.
+	 *
+	 * Ensures the upload_dir filter is fired both when calling this function
+	 * directly and when the upload directory is filtered in the Font Face
+	 * REST API endpoint.
+	 */
+	add_filter( 'upload_dir', 'wp_apply_font_dir_filter' );
```

Comment:
With the need to fallback to the uploads directory this may need to run later (say priority 20) to allow for plugins that filter the uploads directory to an offloader at the default priority, 10. This will avoid an order of operations issue as hooks running at the same priority run in the order they are added.